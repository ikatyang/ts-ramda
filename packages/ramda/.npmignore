node_modules/
scripts/
src/
templates/
.npmignore
test.ts
tsconfig.json
tslint.json
