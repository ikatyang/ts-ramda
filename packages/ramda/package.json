{
  "name": "@ikatyang/types-ramda",
  "version": "0.23.0",
  "description": "TypeScript's type definitions for Ramda",
  "repository": "https://github.com/ikatyang/types",
  "author": "ikatyang",
  "license": "MIT",
  "scripts": {
    "ts-node": "../../node_modules/.bin/ts-node -P ./tsconfig.json",
    "build": "yarn run ts-node -- ./scripts/build.ts ./templates ./typings"
  },
  "devDependencies": {
    "ramda": "0.23.0"
  }
}
