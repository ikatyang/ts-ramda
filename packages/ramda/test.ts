/// <reference path="./index.d.ts" />

import * as R from 'ramda';

R.__;

R.add(1, 2);
R.add(1)(2);
