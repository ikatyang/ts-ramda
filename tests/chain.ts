import * as R from 'ramda';

let duplicate = function(n: number) {
    return [n, n];
};
// @dts-jest
R.chain(duplicate, [1, 2, 3]); // => [1, 1, 2, 2, 3, 3]
// @dts-jest
R.chain(duplicate)([1, 2, 3]); // => [1, 1, 2, 2, 3, 3]
// @dts-jest:skip number[]
R.chain(R.append, R.head)([1, 2, 3]); // => [1, 2, 3, 1]
// @dts-jest:skip number[]
R.chain(R.append)(R.head)([1, 2, 3]); // => [1, 2, 3, 1]
