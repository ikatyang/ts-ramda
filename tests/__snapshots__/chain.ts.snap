// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`R_chain(number_chain_to_string_chain) 1`] = `"<T$1 extends number = number>(list: Chain<T$1>) => Chain<string>"`;

exports[`R_chain(number_chain_to_string_chain)(number_chain) 1`] = `"Chain<string>"`;

exports[`R_chain(number_chain_to_string_chain, number_chain) 1`] = `"Chain<string>"`;

exports[`R_chain(number_to_string_string_tuple) 1`] = `"<T$1 extends number = number>(list: List<T$1>) => string[]"`;

exports[`R_chain(number_to_string_string_tuple)(number_array) 1`] = `"string[]"`;

exports[`R_chain(number_to_string_string_tuple, number_array) 1`] = `"string[]"`;

exports[`R_chain(string_to_number_list_to_object) 1`] = `"<T$1 extends number = number, V$1 extends string = string>(monad: (value: List<T$1>) => V$1) => (value: List<T$1>) => object"`;

exports[`R_chain(string_to_number_list_to_object)(number_list_to_string) 1`] = `"(value: List<number>) => object"`;

exports[`R_chain(string_to_number_list_to_object, number_list_to_string) 1`] = `"(value: List<number>) => object"`;
